### DISCUSSION TITLE
Extension of type 'ErrorProneOptions' does not exist. Currently registered extension types: [ExtraPropertiesExtension, AjcAction]

### DISCUSSION THREAD

**Discussion by nagkumar** [tbroyer/gradle-errorprone-plugin#83] [Q&A] [answered]
FAILURE: Build failed with an exception.

* Where:
Build file 'edu\build.gradle.kts' line: 364

* What went wrong:
Extension of type 'ErrorProneOptions' does not exist. Currently registered extension types: [ExtraPropertiesExtension, AjcAction]

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
> Get more help at https://help.gradle.org.

While using NullAway, we get this error.. any clues

**Comment by nagkumar** [comment]
Changed 
```
options.errorprone.disableWarningsInGeneratedCode.set(true)
    if (!name.lowercase(Locale.getDefault()).contains("test")) {
	configure<ErrorProneOptions> {
	    check("NullAway", CheckSeverity.ERROR)
	    option("NullAway:AnnotatedPackages", "com.tejasoft")
	}
    }
```
to this

tasks.withType<JavaCompile> {
    options.errorprone.disableWarningsInGeneratedCode.set(true)
    if (!name.lowercase(Locale.getDefault()).contains("test"))
    {
	options.errorprone {
	    check("NullAway", CheckSeverity.ERROR)
	    option("NullAway:AnnotatedPackages", "com.tejasoft")
	}
    }
}

It is working now.. documents this way are to be corrected as they give the first snippet..

Any case author pl. confirm the proposed gradle configuration is correct..

**Comment by tbroyer** [comment, marked as answer]
You might want to use my [`net.ltgt.nullaway` plugin](https://github.com/tbroyer/gradle-nullaway-plugin) to make it even easier to configure, but indeed changing `configure<ErrorProneOptions>` to `options.errorprone` is the right thing to do, as the `ErrorProneOptions` extension is added to the `options` object, not the `JavaCompile` task itself.

Further improvements:
* use `withType<JavaCompile>().configureEach {`: https://docs.gradle.org/current/userguide/task_configuration_avoidance.html
* directly configure the `compileJava` task rather than excluding the test tasks

With all those applied, you'd have something like:
```kotlin
nullaway {
  annotatedPackages.add("com.tejasoft")
}
tasks {
  withType<JavaCompile>().configureEach {
    options.errorprone.disableWarningsInGeneratedCode.set(true)
  }
  compileJava {
    options.errorprone.nullaway.error()
  }
}
```
(fwiw, I prefer nesting everything: `options.errorprone { disableWarningsInGeneratedCode.set(true) }`, but YMMV)
